lib/core/error.js

```diff
@@ -152,12 +152,35 @@ function isRetryableError(error) {
   );
 }
 
+const SDAM_RETRYABLE_ERROR_CODES = new Set([
```

Comment:
I think these are more like "errors that SDAM can _not_ recover from", rather than "errors which SDAM can retry on". Maybe `SDAM_UNRECOVERABLE_ERROR_CODES`?